br/pkg/restore/client.go

```diff
@@ -2137,62 +2147,67 @@ func (rc *Client) CleanUpKVFiles(
 // InitSchemasReplaceForDDL gets schemas information Mapping from old schemas to new schemas.
 // It is used to rewrite meta kv-event.
 func (rc *Client) InitSchemasReplaceForDDL(
+	dbMaps []*backuppb.PitrDBMap,
 	tables *map[int64]*metautil.Table,
 	tableFilter filter.Filter,
 ) (*stream.SchemasReplace, error) {
-	dbMap := make(map[stream.OldID]*stream.DBReplace)
-
-	for _, t := range *tables {
-		name, _ := utils.GetSysDBName(t.DB.Name)
-		dbName := model.NewCIStr(name)
-		newDBInfo, exist := rc.GetDBSchema(rc.GetDomain(), dbName)
-		if !exist {
-			log.Info("db not existed", zap.String("dbname", dbName.String()))
-			continue
-		}
+	dbReplaces := make(map[stream.UpstreamID]*stream.DBReplace)
+	if len(dbMaps) <= 0 {
```

Comment:
consider this situation:
1. In the full restore stage, br success to finish.
2. In the log restore stage, br fails.
3. The user drops a table. (But we already has a map that `upstream 80 -> downstream 81`.
4. The user restart the `restore point`.
5. In the full restore stage, br creates the dropped table and success to finish. (Now the downstream table id is 82.
6. In the restore point stage, the `upstream 80` still maps to `downstream 80` instead of `downstream 81`. (ERROR

Reply:
another situation:

1. In the full restore stage, br success to finish.
2. In the log restore stage, br success to finish.
3. The user find that the time range is somewhere mismatch.

They need a new full backup base, so

1. Just drop all the tables, and start to run restore for another full backup data and another range time for log backup data.
2. In the log restore stage, the map is out of date. (ERROR

It seems that here we need to remove the nonexistent downstream table.